lib/Jit/LLILCJit.cpp

```diff
@@ -552,41 +549,61 @@ void ObjectLoadListener::getDebugInfoForObject(
   }
 }
 
-void ObjectLoadListener::recordRelocations(const ObjectFile &Obj) {
+void ObjectLoadListener::recordRelocations(
+    const ObjectFile &Obj, const RuntimeDyld::LoadedObjectInfo &L) {
   for (section_iterator SI = Obj.section_begin(), SE = Obj.section_end();
        SI != SE; ++SI) {
-    if (SI->isText()) {
-      section_iterator RelocatedSection = SI->getRelocatedSection();
+    section_iterator RelocatedSection = SI->getRelocatedSection();
+
+    if (RelocatedSection == SE) {
+      continue;
+    }
+
+    relocation_iterator I = SI->relocation_begin();
+    relocation_iterator E = SI->relocation_end();
+
+    for (; I != E; ++I) {
 
-      if (RelocatedSection == SE) {
+      StringRef SectionName;
+      std::error_code ErrorCode = SI->getName(SectionName);
+      if (ErrorCode) {
+        assert(false && ErrorCode.message().c_str());
+      }
+
+      symbol_iterator Symbol = I->getSymbol();
+      assert(Symbol != Obj.symbol_end());
+
+      SymbolRef::Type SymbolType = Symbol->getType();
+
+      if (SymbolType == SymbolRef::ST_Debug) {
         continue;
       }
 
-      relocation_iterator I = SI->relocation_begin();
-      relocation_iterator E = SI->relocation_end();
-
-      for (; I != E;) {
-        symbol_iterator Symbol = I->getSymbol();
-        assert(Symbol != Obj.symbol_end());
-        section_iterator SecI(Obj.section_end());
-        Symbol->getSection(SecI);
-        const bool IsExtern = SecI == Obj.section_end();
-        if (IsExtern) {
-          // This is an external symbol. Verify that it's one we created for
-          // a global variable and report the relocation via Jit interface.
-          uint64_t RelType = I->getType();
-          uint64_t Offset = I->getOffset();
-          ErrorOr<StringRef> NameOrError = Symbol->getName();
-          assert(NameOrError);
-          StringRef TargetName = NameOrError.get();
-          assert(Context->NameToHandleMap.count(TargetName) == 1);
-          uint64_t Target = Context->NameToHandleMap[TargetName];
-          Context->JitInfo->recordRelocation(MM->getCodeSection() + Offset,
-                                             (void *)Target,
-                                             translateRelocationType(RelType));
-        }
-        ++I;
+      if (!SectionName.compare(".pdata")) {
```

Comment:
Relocation types we get there are not supported by the jit interface's recordRelocation. We won't use the contents of the .pdata section so there is no need to resolve these relocations.